### DISCUSSION TITLE
Playing YouTube videos on iPhone and Android devices with vlitejs

### DISCUSSION THREAD

**Discussion by suryab21** [yoriiis/vlitejs#87] [Q&A] [answered]
Hi @yoriiis, I found this plugin very useful as with common implentation it supports for Video, Audio 
YouTube. I wonder if this plugin also works well on iPhone and Android devices for YouTube. We are planning to integrate this in one of our angular project where we want to create a component which can play the YouTube and other (mp4) videos on desktop as well as on iPhone and Android devices. We are using angular 13. We have seen the documentation which says that some features are not supported on iPhone but does it mean that we cannot even play YouTube video on devices. Our understanding is that as the normal embed code of Youtube works on a html page on devices, Vlitejs will also be able to play YouTube videos on iPhone as well as on Android. Your feedback will be appreciated on this query so that we can plan accordingly.

Many thanks,
Best Wishes

**Comment by yoriiis** [comment, marked as answer]
Hi @suryab21, thanks for your message!

Yes the library works well on desktop and mobile devices (See [browser-support](https://github.com/vlitejs/vlite#browser-support)).
You can check the online demo or the online playgrounds accessible on the README to check if the behavior works for you.

---

There is a limitation for iOS Safari with the fullscreen API on video elements.

https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API#api.document.fullscreen

**Reply by sunilten** [reply]
Hi @yoriiis, 

Continuing to same thread above, API page for vlitejs mentions that some of attributes are not supported for iOS device. 
Please refer to attached screenshot
Below are attribute list as per page which wont work on iOS device. Can you confirm this statement if functionality related to these attributes wont work on iOS devices?
controls¹ ², fullscreen¹,poster¹,bigPlay¹,playsinline¹,muted¹,autoHide¹,autoHideDelay¹

<img width="852" alt="Screenshot 2022-12-02 at 2 57 37 PM" src="https://user-images.githubusercontent.com/119654578/205261014-167c0402-8885-41b2-8f9e-ff208f514953.png">

**Reply by sunilten** [reply]
@yoriiis , Can you help on above. Your answer will help us in determining next steps for our project.

**Reply by yoriiis** [reply]
Hi @sunilten, indeed there is an inconsistency in the documentation, I updated in https://github.com/vlitejs/vlite/commit/104e1929f96cbc3e8492a238513a6674f96139e0.

Prefer to use these options in Vlitejs constructor instead of HTML attributes.
